pipelines/scripts/download_live_private_endpoints.py

```diff
@@ -0,0 +1,22 @@
+from pipelines.scripts.synapse_artifact.synapse_managed_private_endpoint_util import SynapseManagedPrivateEndpointUtil
+import argparse
+import json
+
+
+"""
+Download the live managed private endpoint artifacts from the workspace, and write them locally.
+This is needed to allow the Synapse deployment to work with the below options
+- DeleteArtifactsNotInTemplate: true
+- DeployManagedPrivateEndpoints: false
+"""
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("-e", "--env", required=True, help="The environment to target")
+    args = parser.parse_args()
+    env = args.env
+    synapse_workspace = f"pins-synw-odw-{env}-uks"
+    all_endpoints = SynapseManagedPrivateEndpointUtil(synapse_workspace).get_all(vnet="default")
```

Comment:
Have you considered to add some error handling here ? If the utils fails the script will crash